includes/documents/abstract-wcpdf-order-document.php

```diff
@@ -256,7 +256,7 @@ public function init() {
 
 	public function save( $order = null ) {
 		$order = empty( $order ) ? $this->order : $order;
-		if ( empty( $order ) ) {
+		if ( empty( $order ) || ( is_admin() && ! apply_filters( 'wpo_wcpdf_current_user_is_allowed', current_user_can( 'manage_woocommerce' ), $this->type ) ) ) {
```

Comment:
I'm not sure this is a good idea, you're adding more restrictions to this than existed previously.
Is there a specific reason this is required on top of the check inside the methods that handle AJAX calls?
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/f61d715cf49d0745c90fdb81eb4956763d462107/includes/class-wcpdf-admin.php#L772-L776
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/f61d715cf49d0745c90fdb81eb4956763d462107/includes/class-wcpdf-main.php#L337

Reply:
I've found that if you edit the invoice number in the 'PDF document data' meta box and click the 'Update' button or hit return, the invoice number would still update. So I tried to find a way to also prevent editing document data when not using AJAX. That's why I implemented the filter in the `save()` function.

Reply:
Placing a filter there would only prevent invoice data to be changed, correct? Wouldn't we also want the option to restrict access to other documents?

Reply:
Yes, except that save process is not controlled by the free plugin, so it should be handled (with the same filter) in the plugins controlling those processes

Reply:
Remove filter from core save function: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/commit/094996cebb095eb069d055493aaf3818151ddae1

Add filter to `save_invoice_number_date()`: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/commit/9461baeaca919dbcaac22a23de74cc9668bad93f